db.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	_ "modernc.org/sqlite"
+)
+
+var (
+	errNoMatches   = errors.New("no conversations found")
+	errManyMatches = errors.New("multiple conversations matched the input")
+)
+
+func openDB(path string) (*convoDB, error) {
+	db, err := sqlx.Open(
+		"sqlite",
+		"file://"+filepath.Join(path, "mods.db"),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not create db: %w", err)
+	}
+	if err := db.Ping(); err != nil {
+		return nil, fmt.Errorf("could not ping db: %w", err)
+	}
+	if _, err := db.Exec(`
+		create table if not exists conversations(
+			id string not null primary key,
+			title string not null,
+			updated_at datetime not null default(strftime('%Y-%m-%d %H:%M:%f', 'now')),
+			check(id <> ''),
+			check(title <> '')
+		);
+	`); err != nil {
+		return nil, fmt.Errorf("could not migrate db: %w", err)
+	}
+	return &convoDB{db: db}, nil
+}
+
+type convoDB struct {
+	db *sqlx.DB
+}
+
+// Conversation in the database.
+type Conversation struct {
+	ID        string    `db:"id"`
+	Title     string    `db:"title"`
+	UpdatedAt time.Time `db:"updated_at"`
+}
+
+func (c *convoDB) Close() error {
+	return c.db.Close() //nolint: wrapcheck
+}
+
+func (c *convoDB) Save(id, title string) error {
+	res, err := c.db.Exec(`
+		update conversations
+		set title = $2, updated_at = current_timestamp
+		where id = $1
+	`, id, title)
```

Comment:
Perhaps use `c.db.Rebind()` with question mark query syntax

https://jmoiron.github.io/sqlx/#bindvars

```suggestion
	res, err := c.db.Exec(c.db.Rebind(`
		update conversations
		set title = ?, updated_at = current_timestamp
		where id = ?
	`), title, id)
```